packages/apps/human-app/frontend/src/pages/worker/jobs/components/available-jobs/desktop/available-jobs-table.tsx

```diff
@@ -36,123 +35,133 @@ export type AvailableJobsTableData = AvailableJob & {
+    header: t('worker.jobs.jobType'),
+    size: 200,
+    enableSorting: false,
+    Cell: ({ row }) => {
+      const label = t(`jobTypeLabels.${row.original.job_type as JobType}`);
+      return <Chip label={label} />;
+    },
+    muiTableHeadCellProps: () => ({
+      component: (props) => {
+        return (
           <TableHeaderCell
             {...props}
-            headerText={t('worker.jobs.rewardAmount')}
+            headerText={t('worker.jobs.jobType')}
             iconType="filter"
-            popoverContent={<AvailableJobsRewardAmountSort />}
+            popoverContent={<AvailableJobsJobTypeFilter />}
           />
-        ),
-      }),
-    },
-    {
-      accessorKey: 'job_type',
-      header: t('worker.jobs.jobType'),
-      size: 200,
-      enableSorting: false,
-      Cell: ({ row }) => {
-        const label = t(`jobTypeLabels.${row.original.job_type as JobType}`);
-        return <Chip label={label} />;
+        );
       },
-      muiTableHeadCellProps: () => ({
-        component: (props) => {
-          return (
-            <TableHeaderCell
-              {...props}
-              headerText={t('worker.jobs.jobType')}
-              iconType="filter"
-              popoverContent={<AvailableJobsJobTypeFilter />}
-            />
-          );
+    }),
+  },
+  {
+    accessorKey: 'escrow_address',
+    id: 'selectJobAction',
+    header: '',
+    size: 100,
+    enableSorting: false,
+    Cell: (props) => {
+      const { escrow_address, chain_id } = props.row.original;
+      const { onJobAssignmentError, onJobAssignmentSuccess } =
+        useJobsNotifications();
+      const { mutate: assignJobMutation, status } = useAssignJobMutation(
+        {
+          onSuccess: onJobAssignmentSuccess,
+          onError: onJobAssignmentError,
         },
-      }),
-    },
-    {
-      accessorKey: 'escrow_address',
-      id: 'selectJobAction',
-      header: '',
-      size: 100,
-      enableSorting: false,
-      Cell: (props) => {
-        const { escrow_address, chain_id } = props.row.original;
-        return (
-          <Grid sx={{ display: 'flex', justifyContent: 'flex-end' }}>
-            <Grid sx={{ width: '5rem', height: '2.5rem', padding: '0.2rem 0' }}>
-              <TableButton
-                loading={loadingButtons}
-                onClick={() => {
-                  assignJob({ escrow_address, chain_id });
-                }}
+        [`assignJob-${escrow_address}`]
+      );
+
+      return (
+        <Grid sx={{ display: 'flex', justifyContent: 'flex-end' }}>
+          <Grid sx={{ width: '5rem', height: '2.5rem', padding: '0.2rem 0' }}>
+            <TableButton
+              loading={status === 'pending'}
+              onClick={() => {
+                assignJobMutation({ escrow_address, chain_id });
+              }}
+              sx={{
+                width: '94px',
+              }}
+            >
+              <Typography
+                sx={{ color: 'white !important' }}
```

Comment:
Should it be some constant from common color palette? Why we need this color override here?